neuralhydrology/utils/samplingutils.py

```diff
@@ -219,7 +219,10 @@ def _get_frequency_last_n(self, freq_suffix: str):
         if isinstance(self.predict_last_n, int):
             frequency_last_n = self.predict_last_n
         else:
-            frequency_last_n = self.predict_last_n[freq_suffix[1:]]
+            if freq_suffix != '':
```

Comment:
This and the change below are essentially the same code. No strong feelings but maybe move it to a small function?